### PR TITLE
Refactor + Grass Cure Bug

### PR DESCRIPTION
**Pull request by Sybrand**
- remove broad exception in hfi calc, checking for null values instead.
- removed default fuel_type from fire zone response.
- fix grass cure checks to use configured fuel type (was using default)

# Test Links:
[Percentile Calculator](https://wps-pr-1961.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1961.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1961.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1961.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1961.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN)
[Fire Behaviour Advisory](https://wps-pr-1961.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1961.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-1961.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

**Review by tessmgood — COMMENTED** [review]
I think I found a bug? On first use, when I changed the dates to go back in time a few days/weeks in April to find a missing grass cure, the table loaded with 'no forecast' errors for all stations in Kamloops, which I know is not true. Once I clicked on a daily tab the data did load, when I went back to prep table the errors were cleared, and all date changes now work as expected - just an issue on the initial date change.

**Review by tessmgood — CHANGES_REQUESTED** [review]
not sure if this was introduced in the context of this bug, but it's weird that HFI is displaying as 0.0, but nothing else (ROS, FIG) - they should all be consistent / blank (it's not an HFI of zero, we can't calculate it). Otherwise errors are looking gouda!

Screen Shot 2022-04-28 at 9 49 50 AM

File: web/src/features/hfiCalculator/components/StaticCells.tsx [hunk 1]
```diff
@@ -15,18 +15,20 @@ export interface StaticCellsProps {
   station: WeatherStation
   classNameForRow: string | undefined
   isRowSelected: boolean
+  selectedFuelType: FuelType | undefined
```
[the remark this answers is not in the document]
Comment (Sybrand) [reply web/src/features/hfiCalculator/components/StaticCells.tsx:18]: There should always be a selected fuel type, the back end will always provide a fuel type for a station, and is responsible for putting the default fuel type id in there.
Re. the undefined - I'd prefer it not be undefined, but that would be a bit more of a re-factor in web/src/features/hfiCalculator/pages/HfiCalculatorPage.tsx to make it conditionally only render the ViewSwitcher if the result is defined.
I'll add that onto the re-factor list - but would prefer to make that change in a seperate PR - it affects some of our unit tests, which do create scenarios where there is no fuel type specified.

**Review by andrea-williams — APPROVED** [review]

**Review by tessmgood — APPROVED** [review]
crushed it

**Comment by conbrad** [comment]
Is it possible to add more context to the CFFDRS error messages? Like "`Failed to calculate X, one or more inputs possibly null`?

File: api/app/db/crud/hfi_calc.py [hunk 2]
```diff
@@ -100,8 +100,3 @@ def get_fuel_types(session: Session) -> CursorResult:
 def get_fuel_type_by_id(session: Session, fuel_type_id: int) -> FuelType:
     """ Get the fuel type for the supplied fuel type id """
     return session.query(FuelType).filter(FuelType.id == fuel_type_id).first()
-
-
-def get_fuel_types(session: Session) -> CursorResult:
```
[the remark this answers is not in the document]
Comment (Sybrand) [reply api/app/db/crud/hfi_calc.py:105]: It's a duplicate. I copied the function from your PR while you were working on it - after merge it showed up twice.

**Comment by Sybrand** [comment]
> Is it possible to add more context to the CFFDRS error messages? Like "`Failed to calculate X, one or more inputs possibly null`?

It's a bit fuzzy saying "might be null". For some we can infer it (like ROS for a grass type without curing) - but then we should check for that higher up. We can't say for sure why something in R is not giving us a response.

**Review by conbrad — APPROVED** [review]

**Pull request merged**